### DISCUSSION TITLE
Calculation of `unlit_factor`

### DISCUSSION THREAD

**Discussion by LZhaoM** [valhalla/valhalla#4725] [Q&A] [answered]
Hi,

I am quite curious about the calculation of the `unlit_factor`. Is there any story behind it? What if `unlit_factor_ = unlit_factor + 1.f` is used?
https://github.com/valhalla/valhalla/blob/6fbe96ef28fdea1118972282c5eea9d8a777b281/src/sif/dynamiccost.cc#L365-L368
Thank you

**Comment by chrstnbwnkl** [comment, marked as answer]
Yeah there was a discussion about it when it was introduced in #3957 . The idea was to convert the `use_lit` value (a float between 0 and 1) into a cost factor in a way that does not impact costing when `use_lit == 0`, and also has a stronger impact when `use_lit > 0.5`.

**Reply by LZhaoM** [reply]
Thank you so much for answering! I am trying to introduce a new factor about crime rate. Is there a preferred range for a factor (like the range of `unlit_factor` is `[1,8]`)?

**Reply by chrstnbwnkl** [reply]
It really depends on what you want to achieve. With `unlit_factor` the idea was not too exaggerate the factor too much, since it would not be very sensible to have a huge detour on a route just to get a few meters of lit road.  As stated in the [API reference](https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/#costing-options) a factor can take a max value of 100000, so there's lots of room, but whether or not you need it is completely use case dependent.

**Reply by LZhaoM** [reply]
Thank you so much for the response! I understand it now!
